cmd/cmd_test.go

```diff
@@ -90,3 +91,30 @@ func (test *CommandTest) Teardown(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// terminalOverride lets us more easily redirect streams in the tests.
+type terminalOverride struct {
+	oldOut, oldErr, newOut, newErr io.Writer
+}
+
+// newTerminalOverride creates a new value to override the streams.
+func newTerminalOverride() terminalOverride {
+	return terminalOverride{
+		oldOut: Out,
+		oldErr: Err,
+		newOut: ioutil.Discard,
+		newErr: ioutil.Discard,
+	}
+}
+
+// override sets the package variables to the fake streams.
+func (to terminalOverride) override() {
+	Out = to.newOut
+	Err = to.newErr
+}
```

Comment:
I didn't keep this in the `newTerminalOverride` function for two reasons. First, sometimes we want different behavior than an `ioutil.Discard`, so that lets us set that up manually before we actually trigger the override. Second, I don't like having side effects in a `new` function, especially side effects that affect the entire package.

Reply:
> I don’t like having side effects in a new function, especially side effects that affect the entire package.

That’s a good point. I like the approach! Thanks for calling out the rationale.